upstream-community-operators [N] elastic-phenix-operator (0.1.0)

wmellouli opened the pull request:
Thanks submitting your Operator. Please check below list before you create your Pull Request.

### New Submissions

* [x] Does your operator have [nested directory structure](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#create-a-bundle)?
* [x] Have you selected the Project *Community Operator Submissions* in your PR on the right-hand menu bar?
* [x] Are you familiar with our [contribution guidelines](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md)?
* [ ] Have you [packaged and deployed](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) your Operator for Operator Framework?
* [x] Have you tested your Operator with all Custom Resource Definitions?
* [x] Have you tested your Operator in all supported [installation modes](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#operator-metadata)?
* [x] Have you considered whether you want use [semantic versioning order](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#updating-your-existing-operator)?
* [x] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#sign-your-work)?
* [x] Is operator [icon](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#operator-icon) set?

### Updates to existing Operators

* [ ] Did you create a `ci.yaml` file according to the [update instructions](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#updating-your-existing-operator)?
* [ ] Is your new CSV pointing to the previous version with the `replaces` property if you chose `replaces-mode` via the `updateGraph` property in `ci.yaml`?
* [ ] Is your new CSV referenced in the [appropriate channel](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#bundle-format) defined in the `package.yaml` or `annotations.yaml` ?
* [ ] Have you tested an update to your Operator when deployed via OLM?
* [ ] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#sign-your-work)?

### Your submission should not

* [x] Modify more than one operator
* [x] Modify an Operator you don't own
* [x] Rename an operator - please remove and add with a different name instead
* [x] Submit operators to both `upstream-community-operators` and `community-operators` at once
* [x] Modify any files outside the above mentioned folders
* [x] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [x] Description about the managed Application and where to find more information
2. [x] Features and capabilities of your Operator and how to use it
3. [x] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [x] Human readable name and 1-liner description about your Operator
* [x] Valid [category name](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-required-fields.md#categories)<sup>1</sup>
* [x] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [x] Links to the maintainer, source code and documentation
* [x] Example templates for all Custom Resource Definitions intended to be used
* [x] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

--

<sup>1</sup> If you feel your Operator does not fit any of the pre-defined categories, file an issue against this repo and explain your need

<sup>2</sup> For more information see [here](https://github.com/operator-framework/operator-sdk/blob/master/doc/images/operator-capability-level.svg)

wmellouli commented:
@mvalahtv & @J0zi I got this error in the check step `kiwi / Full operator test`, in `TASK [validate_operator_bundle : Validate the operator bundle manifest and metadata with operator-sdk bundle validate from image kind-registry:5000/test-operator/elastic-phenix-operator:v0.1.0]`:

```
    time="2021-04-01T23:21:19Z" level=error msg="Error: Field spec.conversion.webhookClientConfig.service.port, Value 0: spec.conversion.webhookClientConfig.service.port: Invalid value: 0: port is not valid: must be between 1 and 65535, inclusive"
    time="2021-04-01T23:21:19Z" level=error msg="Error: Field spec.conversion.webhookClientConfig.service.port, Value 0: spec.conversion.webhookClientConfig.service.port: Invalid value: 0: port is not valid: must be between 1 and 65535, inclusive"
```

The field `spec.conversion.webhookClientConfig.service.port` is optional, and default port value is `443`: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#service-reference

I added explicitly port number:
- in `ElasticIndex` CRD: https://github.com/operator-framework/community-operators/pull/3487/files#diff-80b5cfe9e1ac166a4e0be4ba63b2541437860abfd52fdd13785aa2cb8ca93abdR28-R34
- in `ElasticTemplate` CRD: https://github.com/operator-framework/community-operators/pull/3487/files#diff-33d353ebc364b1930898d6dbe307e0982bad49c983e8192a4b0aa5d2f9dda3b5R28-R34

wmellouli commented:
`kiwi / Full operator` test is KO, and I don't know what's wrong:

```
TASK [deploy_olm_operator_upstream_bundle : Operator deployment with OLM error status] ***
fatal: [localhost]: FAILED! => changed=false 
  msg: Operator deployment with OLM failed, check the outputs above for more details
```
Outputs above this error:

```
TASK [deploy_olm_operator_upstream_bundle : Print namespace state] *************
ok: [localhost] => 
  dooub_namespace_state.stdout: |-
    NAME                                                              READY   STATUS              RESTARTS   AGE
    ...
    elastic-phenix-operator-controller-manager-7674f6fdcd-p89d4       0/2     ContainerCreating   0          15m
    ...
```
I see that pod remains in status `ContainerCreating` after 15mn. Can I have more details (e.g. `kubectl describe pod/elastic-phenix-operator-controller-manager-7674f6fdcd-p89d4`  or `kubectl describe deployment/elastic-phenix-operator-controller-manager`) ?

J0zi commented:
@wmellouli, you can run it locally and debug interactively using https://github.com/operator-framework/community-operators/blob/master/docs/operator-test-suite.md.

wmellouli commented:
@J0zi, thank you for your reply.
I used `operator-test-playbooks` to test and debug locally (as suggested), and now `kiwi` test is **OK**.
In fact, I used https://operatorhub.io/packages to generate `ClusterServiceVersion`. But it didn't generate `webhookdefinitions`, `permissions` and `clusterPermissions`, so now I added them manually to fix `kiwi` test.

BTW I added reviewrs as required recently in `ci.yaml` (https://operator-framework.github.io/community-operators/operator-ci-yaml/)

Could you please reply with `/ok-to-test` to run tests.
